src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -23,11 +47,14 @@ export const TodoFilter = () => (
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
         {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
+        {isShowClearButton && (
+          <button
+            data-cy="clearSearchButton"
+            onClick={handleCleanSearch}
```

Comment:
Consider calling `event.preventDefault()` in the `handleCleanSearch` function to prevent any default form submission behavior when the clear button is clicked. This ensures that the form does not inadvertently submit when the button is used.